liquibase-core/src/main/java/liquibase/diff/DiffGeneratorFactory.java

```diff
@@ -84,7 +84,7 @@ public DiffResult compare(Database referenceDatabase, Database comparisonDatabas
 
     public DiffResult compare(DatabaseSnapshot referenceSnapshot, DatabaseSnapshot comparisonSnapshot, CompareControl compareControl) throws DatabaseException {
         Database referenceDatabase = referenceSnapshot.getDatabase();
-        if (comparisonSnapshot !=null && referenceSnapshot!=null) {
+        if (comparisonSnapshot !=null && referenceDatabase!=null) {
```

Comment:
@nvoxland Do we need to do a null check for referenceSnapshot too?

Reply:
No, referenceSnapshot is never null so it doesn't have to be checked. We're already referencing it on line 86 so if it was a problem it would have been seen breaking there sometime which it isn't.